plugin/build.gradle

```diff
@@ -18,10 +20,11 @@ plugins {
     id 'java-gradle-plugin'
     id 'maven-publish'
     id 'com.gradle.plugin-publish' version '0.21.0'
+    id("org.jetbrains.kotlinx.binary-compatibility-validator") version "0.15.0-Beta.1"
+    id("org.jetbrains.kotlin.jvm") version "$kotlin_version"
```

Comment:
For some context: I moved KGP to the `plugins {}` block to try and make the `kotlin {}` configuration block work, but that didn't seem to work in Groovy. Using `plugins {}` for plugin management is preferred anyway, so I kept the change.

I suspect that Gradle should still be able to resolve KGP from the repositories defined inside `buildscript {}`. Is there a way we can check?

I am lining up plans to change this in the near future. When developing a Gradle plugin, it shouldn't use KGP at all, because it might bring in dependencies or add configuration that conflicts with the embedded Kotlin version that Gradle uses at runtime. Instead `embedded-kotlin` or `kotlin-dsl` should be used instead, but I'll make another PR with this change later.

Reply:
> Is there a way we can check?

You can check with: 
`./gradlew clean -Pkotlin_repo_url=https://maven.pkg.jetbrains.space/kotlin/p/kotlin/dev -Pkotlin_version=2.0.0-dev-19952 --i`

Reply:
Gotcha, thanks. I see the problem with my changes now, so I've reverted them.